earthfile2llb/save_image_wait_item.go

```diff
@@ -38,6 +39,6 @@ func (siwi *saveImageWaitItem) SetDoPush() {
 	siwi.mu.Lock()
 	defer siwi.mu.Unlock()
 	if siwi.si.DockerTag != "" {
-		siwi.push = true
+		siwi.doPush = siwi.allowPush
```

Comment:
This is the key change, right?

Reply:
good eyes! yeah we were always setting it to `true` whenever a `SAVE IMAGE` was chained via a `BUILD`, and forgot if the `SAVE IMAGE` had a `hasPushFlag=true`.